sbi/inference/snle/snle_base.py

```diff
@@ -259,13 +241,9 @@ def _train(
             sampler=SubsetRandomSampler(val_indices),
         )
 
-        optimizer = optim.Adam(self._posterior.net.parameters(), lr=learning_rate)
-
-        # If we're retraining from scratch each round, reset the neural posterior
-        # to the untrained copy we made at the start.
-        if retrain_from_scratch_each_round:
-            self._posterior = deepcopy(self._untrained_posterior)
-            optimizer = optim.Adam(self._posterior.net.parameters(), lr=learning_rate)
+        optimizer = optim.Adam(
+            list(self._posterior.net.parameters()), lr=learning_rate,
```

Comment:
why is that necessary? and why was it working before?

Reply:
With the new build function for nets, we do not need to store untrained posterior objects